lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -158,7 +158,9 @@ class DocumentImpl extends NodeImpl {
     }
 
     this._URL = parsed;
-    this._origin = whatwgURL.serializeURLOrigin(parsed);
+    this._origin = whatwgURL.serializeURL(parsed) === "about:blank" && privateData.options.referrer ?
```

Comment:
Instead of using referrer, which is only supposed to impact document.referrer, can you just add a new origin option?

That would also avoid all this parsing and serializing. We could just pass `origin: parentDoc._origin`.

Reply:
good point. done.